util/importers/ldc93s1.py

```diff
@@ -0,0 +1,155 @@
+                    with open(txt_file) as open_txt_file:
+                        original = ' '.join(open_txt_file.read().strip().lower().split(' ')[2:]).replace('.', '')
+                        targets.append(original)
+                target = texts_to_sparse_tensor(targets)
+                for index, next_source in enumerate(sources):
+                    npad = ((0,(n_steps - next_source.shape[0])), (0,0))
+                    sources[index] = np.pad(next_source, pad_width=npad, mode='constant')
+                source = np.array(sources)
+                self._batch_queue.put((source, target))
+
+    def next_batch(self):
+        source, target = self._batch_queue.get()
+        return (source, target, source.shape[1])
+
+    @property
+    def total_batches(self):
+        # Note: If len(_txt_files) % _batch_size != 0, this re-uses initial _txt_files
+        return int(ceil(float(len(self._txt_files)) /float(self._batch_size)))
+
+
+def read_data_sets(graph, data_dir, batch_size, numcep, numcontext, thread_count=1):
+    # Conditionally download data
+    LDC93S1_BASE = "LDC93S1"
+    LDC93S1_BASE_URL = "https://catalog.ldc.upenn.edu/desc/addenda/"
+    local_file = base.maybe_download(LDC93S1_BASE + ".wav", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".wav")
+    _ = base.maybe_download(LDC93S1_BASE + ".txt", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".txt")
+
+    # Conditionally extract LDC93S1 data
+    LDC93S1_DIR = "LDC93S1"
+    _maybe_extract(data_dir, LDC93S1_DIR, local_file, ["dev", "test", "train"])
+
+    # Create dev DataSet
+    train = dev = test = _read_data_set(graph, data_dir, LDC93S1_DIR, "dev", thread_count, batch_size, numcep, numcontext)
+
+    # Create test DataSet
+    ##test = _read_data_set(graph, data_dir, LDC93S1_DIR, "test", thread_count, batch_size, numcep, numcontext)
+
+    # Create train DataSet
+    ##train = _read_data_set(graph, data_dir, LDC93S1_DIR, "train", thread_count, batch_size, numcep, numcontext)
+
+    # Return DataSets
+    return DataSets(train, dev, test)
+
+def _maybe_extract(data_dir, extracted_data, archive, sets):
+    wav_file = archive
+    txt_file = archive.replace(".wav", ".txt")
+    # If data_dir/extracted_data does not exist, extract archive in data_dir
+    if not gfile.Exists(path.join(data_dir, extracted_data)):
+        for set in sets:
+            makedirs(path.join(data_dir, extracted_data, set))
+            shutil.copy(wav_file, path.join(data_dir, extracted_data, set))
+            shutil.copy(txt_file, path.join(data_dir, extracted_data, set))
+
+def _read_data_set(graph, data_dir, extracted_data, data_set, thread_count, batch_size, numcep, numcontext):
+    # Create wav dir
+    wav_dir = path.join(data_dir, extracted_data, data_set)
+
+    # Obtain list of txt files
+    txt_files = glob(path.join(wav_dir, "*.txt"))
+
+    # Let us generate one batch for each thread that is going to run
+    txt_files = [ txt_files[0] for x in xrange(batch_size * 2) ]
```

Comment:
To me this makes no sense.

In the method `_create_files_circular_list()` the threads are enumerating over
```
self._files_circular_list
```
which is a [circular list](https://docs.python.org/2.7/library/itertools.html#itertools.cycle) in which each item is the same. So, this serves no purpose.